gemini/sample-apps/llamaindex-rag/backend/indexing/run_parse_embed_index.py

```diff
@@ -0,0 +1,294 @@
+logger = logging.getLogger(__name__)
+
+
+# Load configuration from config.yaml
+def load_config():
+    config_path = os.path.join(
+        os.path.dirname(__file__), "..", "..", "common", "config.yaml"
+    )
+    with open(config_path, "r") as config_file:
+        return yaml.safe_load(config_file)
+
+
+# Load configuration
+config = load_config()
+
+# Initialize parameters
+PROJECT_ID = config["project_id"]
+LOCATION = config["location"]
+INPUT_BUCKET_NAME = config["input_bucket_name"]
+DOCSTORE_BUCKET_NAME = config["docstore_bucket_name"]
+INDEX_ID = config["index_id"]
+VECTOR_INDEX_NAME = config["vector_index_name"]
+INDEX_ENDPOINT_NAME = config["index_endpoint_name"]
+INDEXING_METHOD = config["indexing_method"]
+CHUNK_SIZES = config["chunk_sizes"]
+EMBEDDINGS_MODEL_NAME = config["embeddings_model_name"]
+APPROXIMATE_NEIGHBORS_COUNT = config["approximate_neighbors_count"]
+BUCKET_PREFIX = config["bucket_prefix"]
+VECTOR_DATA_PREFIX = config["vector_data_prefix"]
+CHUNK_SIZE = config.get("chunk_size", 512)
+CHUNK_OVERLAP = config.get("chunk_overlap", 50)
+DOCAI_LOCATION = config["docai_location"]
+DOCAI_PROCESSOR_DISPLAY_NAME = config["document_ai_processor_display_name"]
+DOCAI_PROCESSOR_ID = config.get("docai_processor_id")
+CREATE_DOCAI_PROCESSOR = config.get("create_docai_processor", False)
+FIRESTORE_DB_NAME = config.get("firestore_db_name")
+FIRESTORE_NAMESPACE = config.get("firestore_namespace")
+QA_INDEX_NAME = config.get("qa_index_name")
+QA_ENDPOINT_NAME = config.get("qa_endpoint_name")
+
+
+def main():
+    # Initialize Vertex AI and create index and endpoint
+    aiplatform.init(project=PROJECT_ID, location=LOCATION)
+
+    # Creating Vector Search Index
+    vs_index, vs_endpoint = get_or_create_existing_index(
+        VECTOR_INDEX_NAME, INDEX_ENDPOINT_NAME, APPROXIMATE_NEIGHBORS_COUNT
+    )
+
+    # Vertex Vector Search Vector DB and Firestore Docstore
+    vector_store = VertexAIVectorStore(
+        project_id=PROJECT_ID,
+        region=LOCATION,
+        index_id=vs_index.name,  # Use .name instead of .resource_name
+        endpoint_id=vs_endpoint.name,  # Use .name instead of .resource_name
+        gcs_bucket_name=DOCSTORE_BUCKET_NAME,
+    )
+
+    docstore = FirestoreDocumentStore.from_database(
+        project=PROJECT_ID, database=FIRESTORE_DB_NAME, namespace=FIRESTORE_NAMESPACE
+    )
+
+    # Setup embedding model and LLM
+    embed_model = VertexTextEmbedding(
+        model_name=EMBEDDINGS_MODEL_NAME, project=PROJECT_ID, location=LOCATION
+    )
+    llm = Vertex(model="gemini-1.5-flash", temperature=0.0)
+    Settings.llm = llm
+    Settings.embed_model = embed_model
+
+    # Initialize Document AI parser
+    GCS_OUTPUT_PATH = f"gs://{DOCSTORE_BUCKET_NAME}/{VECTOR_DATA_PREFIX}/docai_output/"
+
+    parser = DocAIParser(
+        project_id=PROJECT_ID,
+        location=DOCAI_LOCATION,
+        processor_name=f"projects/{PROJECT_ID}/locations/{DOCAI_LOCATION}/processors/{DOCAI_PROCESSOR_ID}",
```

Comment:
There is a method on the client library to construct this path. It's probably also preferred to construct this path in the `DocAIParser` class so it doesn't need to be filled all the way every time it's used.

Reply:
this appears only to be possible when a processor already exists. But when creating a new one, you need a fully specified path for where it will live. https://cloud.google.com/document-ai/docs/create-processor